python/lsst/ts/salobj/sal_info.py

```diff
@@ -644,38 +651,343 @@ async def _run_kafka(self) -> None:
         finally:
             await self.close(cancel_run_kafka_task=False)
 
+    def _blocking_setup_kafka(self) -> None:
```

Comment:
Are you sure about prepending "`_blocking`" to these methods names? I think you are trying to highlight the fact that these are regular method and not coroutines and, as such, would "block" the event loop. 

I would suggest removing "_blocking" from all these methods names. You could make a note on the docstring about it, but, since they are not coroutines, I think it is pretty self evident.

Reply:
The point of "_blocking" is that these routines should be run in a thread, because they may take a long time and they do not release the event loop.

That is very unusual and I consider it a crucial distinction. 

Most synchronous methods in salobj are fast enough that they can safely be used in the event loop.